site/in-dev/client_config.md

```diff
@@ -178,40 +210,116 @@ nessie.server.authentication.enabled=true
+
+All the properties required for `client_credentials` are also required for this grant type. As
+explained above, if `nessie.authentication.oauth2.issuer-url` is provided, then no further
+configuration is required. Otherwise, in addition to the token endpoint, the authorization endpoint
+must also be provided (`nessie.authentication.oauth2.auth-endpoint`).
 
 If the terminal session is running remotely however, on inside an embedded device, then the
-"authorization_code" grant type may not be suitable, as the browser and the terminal session must 
-be running on the same machine. In this case, the "device_code" grant type is recommended. Similar 
-to the "authorization_code" grant type, it requires the user to authenticate in a browser window, 
-but it does not require the browser and the terminal session to be running on the same machine. The 
-user will be prompted to authenticate in a local browser window, and the remote Nessie client will
-poll the OAuth2 server for the authentication status, until the authentication is complete.
+`authorization_code` grant type may not be suitable, as the browser and the terminal session must be
+running on the same machine. In this case, the `device_code` grant type is recommended. Similar to
+the `authorization_code` grant type, it requires the user to authenticate in a browser window, but
+it does not require the browser and the terminal session to be running on the same machine. The user
+will be prompted to authenticate in a local browser window, and the remote Nessie client will poll
+the OAuth2 server for the authentication status, until the authentication is complete. 
+
+All the properties required for `client_credentials` are also required for this grant type. As
+explained above, if `nessie.authentication.oauth2.issuer-url` is provided, then no further
+configuration is required. Otherwise, in addition to the token endpoint, the device authorization
+endpoint must also be provided (`nessie.authentication.oauth2.device-auth-endpoint`).
+
+Finally, the `token_exchange` grant type is the most complex one. In-depth configuration of a token
+exchange grant is outside the scope of this document but in general, two use cases can be envisaged:
+
+1. Initial token exchange: enabled when `authentication.oauth2.grant-type` is `token_exchange`. In
+   this scenario, the client will use token exchange as the primary grant. A subject token must be
+   provided with `nessie.authentication.oauth2.token-exchange.subject-token`. Other properties under
+   `nessie.authentication.oauth2.token-exchange.*` may also be required.
+
+2. Impersonation or delegation: this is the most typical usage, enabled when
+   `nessie.authentication.oauth2.impersonation.enabled` is `true`. Here, the client will first
+   obtain an initial token using another grant type, then exchange the received access token for
+   another access token, possibly from a second OAuth2 server. If a second OAuth2 server must be
```

Comment:
nit: It might be worth mentioning that the admin user is responsible for configuring the trust relationship between the two OAuth2 servers (for the sake of clarity).

Reply:
https://github.com/projectnessie/nessie/pull/8928